hera_cal/chunker.py

```diff
@@ -0,0 +1,144 @@
+        name of outputfile to write time-concatenated data too.
+    polarizations: list of strs, optional
+        Limit output to polarizations listed.
+        Default None selects all polarizations.
+    spw_range: 2-list or 2-tuple
+        optional lower and upper channel range to select
+    throw_away_flagged_bls: bool, optional
+        if true, throw away baselines that are fully flagged.
+        default is False.
+    clobber: bool, optional
+        if true, overwrite any preexisting output files.
+        defualt is false.
+    flag_yaml : str, optional
+        yaml file with list of antennas to flag and throw away if throw_away_flagged_bls is True
+    Returns
+    -------
+    Concatenated HERAData object.
+
+    """
+    filenames = sorted(filenames)
+    start = filenames.index(inputfile)
+    end = start + chunk_size
+    hd = io.HERAData(filenames[start:end])
+    read_args = {}
+    if polarizations is not None:
+        read_args['polarizations'] = polarizations
+    if spw_range is not None:
+        read_args['freq_chans'] = np.arange(spw_range[0], spw_range[1]).astype(int)
+    data, flags, nsamples = hd.read(axis='blt', **read_args)
+    # throw away fully flagged baselines.
+    if throw_away_flagged_bls:
+        from hera_qm.utils import apply_yaml_flags
+        hd = apply_yaml_flags(hd, ant_flag_yaml, flag_freqs=False, flag_times=False,
+                              flag_ants=True, ant_indices_only=True, throw_away_flagged_ants=True)
+
+    hd.write_uvh5(outputfile, clobber=clobber)
+
+    return hd
+
+
+def chunk_cal_files(filenames, inputfile, outputfile, chunk_size, spw_range=None, clobber=False):
+    """A calibration file chunker
+
+    Parameters
+    ----------
+    filenames: list of strings
+        list of filenames to chunk. Should be homogenous in blt.
+    inputfile: str
+        name of intput file to start chunk at.
+    outpufile: str
+        name of outputfile to write time-concatenated data too.
+    chunk_size: int
+        number of files after inputfile to include in chunk
+    spw: 2-tuple or 2-list of ints
+        channels to keep. Default, None, keeps all channels.
+    clobber: bool, optional
+        if true, overwrite any preexisting output files.
+        defualt is false.
+    Returns
+    -------
+    Concatenated HERACal object.
+
+    """
+    filenames = sorted(filenames)
+    start = filenames.index(inputfile)
+    end = start + chunk_size
+    hc = io.HERACal(filenames[start:end])
+    hc.read()
+    # throw away fully flagged baselines.
+    if spw_range is not None:
+        hc.select(freq_chans=np.arange(spw_range[0], spw_range[1]).astype(int))
+    hc.write_calfits(outputfile, clobber=clobber)
+    return hc
+
+
+def chunk_cal_parser():
+    """
+    An argument parser for cal chunking.
+
+    Parameters
+    ----------
+    N/A
+
+    Returns
+    -------
+    Argument parser.
+    """
+    a = argparse.ArgumentParser(description="Chunk calibration files.")
```

Comment:
As in #681, we should avoid single-letter variable names when we can.

Reply:
fixed. Now named ap.